tests/test_functional.py

```diff
@@ -354,12 +316,14 @@ def setup(self):
 
     def teardown(self):
         session.rollback()
+        session.expunge_all()
         metadata.drop_all()
 
     def _create_one_lake(self):
         lake = Lake(WKTElement('LINESTRING(0 0,1 1)', srid=4326))
         session.add(lake)
         session.flush()
+        session.expunge_all()
```

Comment:
@elemoine , @zzzeek I had to add several `session.expunge_all()` calls to make the tests pass in PyPy... but I don't understand why it is required now while it was not before. Do you have any clue?

Reply:
I have no clue unfortunately. I see nothing in your changes that could have an impact on that.

Reply:
@adrien-berchet I've just added a commit that removes the call to `expunge_all`, in favor to calls to `expire`, which, I think makes more sense, and `expire` is what's used elsewhere in this test file.